Empty network validation missing: plugin crashes with `panic: interface conversion: interface {} is nil, not map[string]interface {}`

# Bug report

### Description

Empty networks are not validated as `terraform plan` success.
After that, running `terraform apply` crashes. See "Panic Output".

### Terraform Version
<!---
Please run `terraform -v` to show the **CloudSigma provider version** as well
as the **Terraform core version**.

If you are not running the latest version of Terraform or the provider, please
upgrade because your issue may have already been fixed.
--->

```
terraform -v
Terraform v1.5.7
on darwin_arm64
+ provider registry.terraform.io/cloudsigma/cloudsigma v1.11.0
```

### Affected Resource(s)
<!--- Please list the affected resources and data sources. --->

* cloudsigma_server

### Terraform Configuration Files
<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
resource "cloudsigma_server" "test_drive" {
  cpu          = 2000              # 2GHz CPU
  memory       = 512 * 1024 * 1024 # 512MB RAM
  name         = "web"

  drive {
    uuid = cloudsigma_drive.test_new_drive_system.id
  }
  network {
    vlan_uuid = ""
  }
}

# CloudSigma drives
resource "cloudsigma_drive" "test_new_drive_system" {
  media = "disk"
  name  = "test_mario_system"
  size  = 15 * 1024 * 1024 * 1024 # 15GB
  clone_drive_id = data.cloudsigma_library_drive.cloudsigma_library_drives.id
}
```

### Expected Behavior

`terraform plan` fails telling that there is an empty network definition.
<!--- What should have happened? --->

### Actual Behavior
<!--- What actually happened? --->
plugins crashed with panic.

### Steps to Reproduce
<!--- Please list the steps required to reproduce the issue. --->

1. `terraform plan`
1. `terraform apply`

### Debug Output
<!---
Please provide a link to a GitHub Gist containing the complete debug output.
Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, define the `TF_LOG=debug` environment variables
before running `terraform apply`.
--->

### Panic Output
<!--- If Terraform produced a panic, please provide a link to a GitHub Gist
containing the output of the `crash.log`. --->

```console
cloudsigma_server.test_drive: Creating...
╷
│ Error: Request cancelled
│
│   with cloudsigma_server.test_drive,
│   on main.tf line 11, in resource "cloudsigma_server" "test_drive":
│   11: resource "cloudsigma_server" "test_drive" {
│
│ The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵

Stack trace from the terraform-provider-cloudsigma_v1.11.0 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 119 [running]:
github.com/cloudsigma/terraform-provider-cloudsigma/cloudsigma.resourceCloudSigmaServerCreate({0x10323c190, 0x140003ec700}, 0x14000410680, {0x103190380?, 0x1400023ee10?})
        github.com/cloudsigma/terraform-provider-cloudsigma/cloudsigma/resource_cloudsigma_server.go:186 +0x1698
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x14000171c00, {0x10323c0e8, 0x14000419860}, 0xd?, {0x103190380, 0x1400023ee10})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/resource.go:778 +0xe8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x14000171c00, {0x10323c0e8, 0x14000419860}, 0x140004489c0, 0x14000410500, {0x103190380, 0x1400023ee10})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/resource.go:909 +0x86c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x14000357bc0, {0x10323c0e8?, 0x14000419770?}, 0x14000430410)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/grpc_provider.go:1074 +0xb08
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x14000266dc0, {0x10323c0e8?, 0x14000418300?}, 0x140003ec000)
        github.com/hashicorp/terraform-plugin-go@v0.20.0/tfprotov5/tf5server/server.go:859 +0x3c0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x10320b620?, 0x14000266dc0}, {0x10323c0e8, 0x14000418300}, 0x14000410000, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.20.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:503 +0x164
google.golang.org/grpc.(*Server).processUnaryRPC(0x140002581e0, {0x10323c0e8, 0x14000418270}, {0x103240220, 0x1400041a1a0}, 0x140001b2000, 0x14000366f90, 0x103721498, 0x0)
        google.golang.org/grpc@v1.60.0/server.go:1372 +0xb8c
google.golang.org/grpc.(*Server).handleStream(0x140002581e0, {0x103240220, 0x1400041a1a0}, 0x140001b2000)
        google.golang.org/grpc@v1.60.0/server.go:1783 +0xc4c
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.60.0/server.go:1016 +0x5c
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 36
        google.golang.org/grpc@v1.60.0/server.go:1027 +0x138

Error: The terraform-provider-cloudsigma_v1.11.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/)
  to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

--- a/cloudsigma/resource_cloudsigma_server.go
+++ b/cloudsigma/resource_cloudsigma_server.go
@@ -204,6 +204,9 @@ func resourceCloudSigmaServerCreate(ctx context.Context, d *schema.ResourceData,
 		createRequest.Servers[0].NICs = make([]cloudsigma.ServerNIC, len(networks))
 
 		for i, n := range networks {
+			if n == nil {
+				return diag.Errorf("defined network argument is empty")
+			}
 			network := n.(map[string]interface{})
 			networkType := network["type"].(string)
 			networkAddress := network["ipv4_address"].(string)

